### PR TITLE
refactor: inline all component variants [BAU-59]

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

Inline all component variants instead of picking and excluding from the shared type.

## PR Checklist

- [ ] I have read the relevant `readme.md` file(s)
- [ ] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Storybook stories are added/updated/not required
- [ ] Usage notes are added/updated/not required
- [ ] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [ ] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/button/src/types.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-
 export type ButtonSize = 'small' | 'medium' | 'large';
 
 export type ButtonVariant =
-  | Exclude<ComponentVariant, 'warning'>
+  | 'negative'
+  | 'positive'
+  | 'primary'
+  | 'secondary'
+  | 'warning'
```
Comment (bgutsol) [inline packages/components/button/src/types.ts:8]: Forgot to exclude `warning`

Comment (suevalov) [reply packages/components/button/src/types.ts]: 🦅

File: packages/components/icon/src/Icon.tsx [hunk 2]
```diff
@@ -24,7 +23,15 @@ export type IconComponent = ExoticComponent<any> | ComponentType<any>;
 
 export type IconSize = 'large' | 'medium' | 'small' | 'tiny';
 
-export type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+export type IconVariant = Simplify<
```
Comment (bgutsol) [inline packages/components/icon/src/Icon.tsx:26]: Is there any reason to keep this `Simplify` here?

File: packages/components/spinner/src/Spinner.tsx [hunk 3]
```diff
@@ -14,9 +12,7 @@ import type {
 
 const DEFAULT_TAG = 'div';
 
-export type SpinnerVariant = Simplify<
-  PickUnion<ComponentVariant, 'primary'> | 'default' | 'white'
->;
+export type SpinnerVariant = Simplify<'primary' | 'default' | 'white'>;
```
Comment (bgutsol) [inline packages/components/spinner/src/Spinner.tsx:15]: Is there any reason to keep this `Simplify` here?

**Review by burakukula — APPROVED** [review]

**Review by bgutsol — APPROVED** [review]

**Pull request merged**